src/haddock/libs/libcns.py

```diff
@@ -25,45 +27,94 @@ def generate_default_header(protonation=None):
+        raise TypeError(emsg)
+
+
 def load_workflow_params(default_params):
-    """Write the values at the header section."""
-    param_header = f'{linesep}! Parameters{linesep}'
+    """
+    Write the values at the header section.
 
-    for param, v in default_params.items():
+    "Empty variables" are ignored. These are defined accoring to
+    :func:`inser t_in_CNS`.
 
-        if isinstance(v, bool):
-            v = str(v).lower()
-            param_header += f'eval (${param}={v}){linesep}'
+    Parameters
+    ----------
+    default_params : dict
+        Dictionary containing the key:value pars for the parameters to
+        be written to CNS. Values cannot be of dictionary type.
 
-        elif isinstance(v, str):
-            param_header += f'eval (${param}="{v}"){linesep}'
+    Returns
+    -------
+    str
+        The string with the CNS parameters defined.
+    """
+    param_header = f'{linesep}! Parameters{linesep}'
+
+    non_empty_parameters = (
+        (k, v)
+        for k, v in default_params.items()
+        if insert_in_CNS(v)
+        )
 
-        elif isinstance(v, int):
-            param_header += f'eval (${param}={v}){linesep}'
+    # types besides the ones in the if-statements should not enter this loop
+    for param, v in non_empty_parameters:
+        param_header += write_eval_line(param, v)
 
-        elif isinstance(v, float):
-            param_header += f'eval (${param}={v}){linesep}'
+    return param_header
 
-        elif not v:
-            # either 0 or empty string
-            if isinstance(v, str):
-                v = '\"\"'
-                param_header += f'eval (${param}={v}){linesep}'
-            if isinstance(v, int):
-                v = 0.0
-                param_header += f'eval (${param}={v}){linesep}'
 
-    if 'chain' in default_params:
-        # load molecule specific things
-        for mol in default_params['chain']:
-            for param in default_params['chain'][mol]:
-                v = default_params['chain'][mol][param]
-                # this are LOGICAL, which means no quotes
-                param_header += f'eval (${param}_{mol}={v}){linesep}'
+def load_input_mols(mols):
+    """Load input molecules as defined by the topoaa/defaults.cfg."""
+    param_header = ''
+    for mol, params in mols.items():
+        for param, value in params.items():
+            param_header += write_eval_line(f'{param}_{mol}', value)
 
     return param_header
 
 
+def write_eval_line(param, value):
+    """Write the CNS eval line depending on the type of `value`."""
+    if isinstance(value, bool):
+        value = str(value).lower()
+        return f'eval (${param}={value}){linesep}'
+
+    elif isinstance(value, str):
+        return f'eval (${param}="{value}"){linesep}'
+
+    elif isinstance(value, Path):
+        return f'eval (${param}="{str(value)}"){linesep}'
+
+    elif isinstance(value, (int, float)):
+        return f'eval (${param}={value}){linesep}'
```

Comment:
This will have all decimals of the float, should we do something about it to make it CNS-safe?